### PR TITLE
[ci] Update docker sonic slave pipeline to build slave base docker

### PR DESCRIPTION
**Pull request by liushilongbuaa**
#### Why I did it
Now azure pipeline is using ACR as docker registry instead of docker hub.
It caused change of Dockerfile and hash value. So, DOCKER_BASE_PULL=y didn't work anymore.
#### How I did it
1. Using make showtag to get hash value.
2. Using the same configure variables with official and PR build. (ex: reproducible var, ACR var)
3. Build march image and regular image in same pipeline.
#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205

### CODE REVIEW

File: .azure-pipelines/docker-sonic-slave-template.yml [hunk 1]
```diff
@@ -32,89 +35,58 @@ parameters:
...
+          args='NOJESSIE=1 NOSTRETCH=1 NOBULLSEYE=1'
+          ;;
+        stretch)
+          args='NOJESSIE=1 NOBUSTER=1 NOBULLSEYE=1'
+          ;;
+        jessie)
+          args='NOSTRETCH=1 NOBUSTER=1 NOBULLSEYE=1'
+          ;;
+      esac
+      image_tag=$(make showtag PLATFORM=generic PLATFORM_ARCH=${{ parameters.arch }} $args | grep sonic-slave | tail -n 1)
```
Comment (xumia) [inline .azure-pipelines/docker-sonic-slave-template.yml:71]: How about to replace the linke 57~71 to one line as below?
image_tag=$(BLDENV=${{ parameters.dist }} PLATFORM=generic make -f Makefile.work showtag | grep sonic-slave | tail -n 1)

File: .azure-pipelines/docker-sonic-slave.yml [hunk 2]
```diff
@@ -12,26 +12,27 @@ resources:
     endpoint: sonic-net
 
 schedules:
-- cron: "0 8 * * *"
+- cron: "0 0 * * 0"
```
Comment (xumia) [inline .azure-pipelines/docker-sonic-slave.yml:15]: Can we use batch build, if any change of the slave Dockerfile, we would like to build the new version of the docker image.

File: .azure-pipelines/docker-sonic-slave-template.yml (hunk 1 above)
Comment (liushilongbuaa) [reply .azure-pipelines/docker-sonic-slave-template.yml]: It makes sense.

File: .azure-pipelines/docker-sonic-slave.yml (hunk 2 above)
Comment (liushilongbuaa) [reply .azure-pipelines/docker-sonic-slave.yml:15]: added

**Review by xumia — APPROVED** [review]

**Pull request merged**